src/rewards/voters.rs

```diff
@@ -33,14 +37,65 @@ pub fn calculate_stake<'address>(
 pub fn calculate_reward_share<'address>(
     total_stake: u64,
     stake_per_voter: &HashMap<&'address Address, u64>,
+    threshold_addresses: &AddressesVoteCount,
+    threshold: u64,
 ) -> HashMap<&'address Address, U64F64> {
     stake_per_voter
         .iter()
-        .map(|(k, v)| (*k, fixed::types::U64F64::from_num(*v) / total_stake as u128))
+        .map(|(k, v)| {
+            let reward = if *threshold_addresses
+                .get(k)
+                .unwrap_or_else(|| panic!("Missing address {} in vote count map", k))
+                >= threshold
+            {
+                U64F64::from_num(*v) / total_stake as u128
+            } else {
+                U64F64::default()
+            };
+            (*k, reward)
+        })
         .collect()
 }
 
 /// get the proportional reward from a share and total rewards amount
 pub fn reward_from_share(share: U64F64, total_reward: u64) -> fixed::types::U64F64 {
     fixed::types::U64F64::from_num(total_reward) * share
 }
+
+pub type VoteCount = HashMap<String, u64>;
+pub type AddressesVoteCount = HashMap<Address, u64>;
+
+pub fn vote_count_with_addresses(
+    vote_count: VoteCount,
+    block0: &Block0Configuration,
+) -> AddressesVoteCount {
+    let discrimination = block0.blockchain_configuration.discrimination;
+    vote_count
+        .into_iter()
+        .map(|(account_hex, count)| {
+            (
+                account_hex_to_address(account_hex, discrimination)
+                    .expect("Valid hex encoded UnspecifiedAccountIdentifier"),
+                count,
+            )
+        })
+        .collect()
+}
+
+fn account_hex_to_address(
+    account_hex: String,
+    discrimination: Discrimination,
+) -> Result<Address, hex::FromHexError> {
+    let mut buffer = [0u8; 32];
+    hex::decode_to_slice(account_hex, &mut buffer)?;
+    let identifier: UnspecifiedAccountIdentifier = UnspecifiedAccountIdentifier::from(buffer);
+    Ok(Address::from(chain_addr::Address(
+        discrimination,
+        Kind::Account(
+            identifier
+                .to_single_account()
+                .expect("Only single accounts are supported")
+                .into(),
+        ),
+    )))
```

Comment:
I'm a bit concern about this. I would like to strongly review it.

Reply:
For voting we imply that the input is always of account type, so I think it's fine there.